### PR TITLE
feat: ✨ adds active period filter to CANs list

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

- adds active period filter to CANs list
- does not include tags

## Issue

#347 #2888

## How to test

1. goto CANs list
2. click on filter button
3. select an active period
4. see table update with filter
5. 💰 profit

## Screenshots

https://github.com/user-attachments/assets/ec990df8-6101-4124-84ef-03d98d05d7b1

## Definition of Done Checklist
- [X] OESA: Code refactored for clarity
- [X] OESA: Dependency rules followed
- [X] Automated unit tests updated and passed
- [X] Automated integration tests updated and passed
- [X] Automated quality tests updated and passed
- [X] Automated load tests updated and passed
- [X] Automated a11y tests updated and passed
- [X] Automated security tests updated and passed
- [X] 90%+ Code coverage achieved
- [-] Form validations updated

### CODE REVIEW

File: frontend/src/components/CANs/CANActivePeriodComboBox/CANActivePeriodComboBox.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    ];
+
+    return (
+        <div className="display-flex flex-justify">
+            <div>
+                <label
+                    className={legendClassname}
+                    htmlFor="can-active-period-combobox-input"
+                >
+                    Fiscal Year
```
Comment (jonnalley) [inline frontend/src/components/CANs/CANActivePeriodComboBox/CANActivePeriodComboBox.jsx:40]: @fpigeonjr according to figma design in the For Devs page, this label is supposed to say "Active Period"

Screenshot 2024-10-08 at 02 12 49

**Comment by jonnalley** [comment]
📈

File: frontend/src/components/CANs/CANActivePeriodComboBox/CANActivePeriodComboBox.jsx (hunk 1 above)
Comment (fpigeonjr) [reply frontend/src/components/CANs/CANActivePeriodComboBox/CANActivePeriodComboBox.jsx]: thanks for pointing this out

**Review by rajohnson90 — APPROVED** [review]

File: frontend/src/components/CANs/CANTable/CANTable.jsx [hunk 2]
```diff
@@ -23,7 +19,7 @@ const CANTable = ({ cans, fiscalYear }) => {
     // Filter CANs by fiscal year
     const filteredCANsByFiscalYear = React.useMemo(() => {
         if (!fiscalYear) return cans;
-        return cans.filter((can) => can.funding_budgets.some((budget) => budget.fiscal_year === fiscalYear));
+        return cans.filter((can) => can.funding_details.fiscal_year === fiscalYear);
```
Comment (fpigeonjr) [inline frontend/src/components/CANs/CANTable/CANTable.jsx:22]: thx @johndeange for calling this out

**Review by maiyerlee — APPROVED** [review]
👏👏👏

**Pull request merged**